app/models/result.rb

```diff
@@ -20,96 +16,8 @@ def initialize(title, url)
   def getit_url
-    if marc_856 && relevant_marc_856?
-      best_link(marc_856, 'marc_856')
-    elsif fulltext_links && relevant_fulltext_links?
-      best_link(fulltext_links_picker(true), 'eds fulltext')
-    elsif Flipflop.enabled? :primo_search
-      alma_link_resolver_url
-    end
-  end
-
-  def alma_link_resolver_url
-    self.openurl
-  end
-
-  # URL to use for check_sfx button in the UI
-  def check_sfx_url
-    fulltext_links_picker(false) if fulltext_links && relevant_fulltext_links?
-  end
-
-  def best_link(link, winner)
-    self.winner = winner
-    link
-  end
-
-  # ensure we care about the included marc_856 link
-  def relevant_marc_856?
-    relevant_links.map { |x| marc_856.include?(x) }.any?
-  end
-
-  # domains we consider relevant when evaluating links
-  def relevant_links
-    ['libproxy.mit.edu', 'library.mit.edu', 'sfx.mit.edu', 'owens.mit.edu',
-     'libraries.mit.edu']
-  end
-
-  # Check fulltext_links for specific parameters to allow for prioritization
-  def relevant_fulltext_links?
-    fulltext_links.map do |link|
-      relevant_links.map { |x| link['Url'].include?(x) }.any?
-    end.include?(true)
-  end
-
-  # Exclude any links that include SFX link (not subscribed resources)
-  # as a link['Name']
-  def remove_not_subscribed_sfx_links(links)
-    links.map do |l|
-      l unless l['Name'].present? && l['Name'].include?('not subscribed')
-    end
-  end
-
-  # Only use links going through sfx or library as other direct links
-  # may be to less useful things (like t.o.c.) or direct to publishers, which
-  # while useful on campus, would not be useful off campus without adding
-  # additional features to bento that are currently out of scope.
-  # `subscribed` is a boolean to signal whether to only include sfx links that
-  # are highly likely to go directly to full text (i.e. we know we subscribe
-  # to them or not)
-  def fulltext_links_picker(subscribed)
-    link = fulltext_links.map do |l|
-      relevant_links.map { |x| l if l['Url'].include?(x) }.compact
-    end
-    links = if subscribed
-              remove_not_subscribed_sfx_links(link.flatten)
-            else
-              link.flatten
-            end
-    fulltext_link_sorter(links).first if links
-  end
-
-  # prioritizes proxied links
-  # this is really hack and only moves a specified match to the top and does
-  # not allow for an actual custom sorted prioritization of links
-  def fulltext_link_sorter(link)
-    urls = link.flatten.compact.map { |x| x['Url'] }
-    urls.sort do |x|
-      if x.include?(relevant_links[0])
-        -1
-      else
-        1
-      end
-    end
-  end
-
-  # Reformat the Accession Number to match the format used in Aleph
-  def clean_an
-    if aleph_record?
-      an.split('.').last.prepend('MIT01')
-    elsif aleph_cr_record?
-      an.split('.').last.prepend('MIT30')
-    elsif alma_record?
-      an
+    if Flipflop.enabled? :primo_search
```

Comment:
Matt and I discussed it while you were on vacation. The rationale for keeping it is essentially to ease migrations to future discovery backends. I.e., when we migrate from Primo to Catalogtron 4000, we will already have the Primo feature flag in place that we can switch off when we're ready to launch. It's an unlikely scenario that we'll still be using this Bento app at that point, but it seemed relatively harmless to me to leave the feature flag.

This is not a strongly held opinion, and I'm happy to remove the flag if you think it's more confusing than potentially useful. I should note that this and debug mode are the last two feature flags, which prompts the question of whether we should remove Flipflop altogether.

Reply:
Thinking about this further, it might not make a lot of sense to have a feature flag that, when disabled, functionally breaks the app. It's probably more reasonable to introduce a new feature flag for whatever the next catalog is (assuming this app lasts that long).

I would say, though, that along with removing the feature flag, we should probably also remove Flipflop. Does that make sense?

Reply:
Yeah let's leave it in for now and reconsider in a few months as we start to get a better feel for how our Discovery Roadmap is looking.